File: src/main/java/com/pubnub/api/models/consumer/pubsub/PNMessageResult.java
package com.pubnub.api.models.consumer.pubsub;

import com.google.gson.JsonElement;
import com.pubnub.api.SpaceId;
import lombok.Getter;
import lombok.ToString;

@Getter
@ToString(callSuper = true)
public class PNMessageResult extends MessageResult {
    public PNMessageResult(BasePubSubResult basePubSubResult, JsonElement message, String type, SpaceId spaceId) {
        super(basePubSubResult, message, type, spaceId);
    }
}

File: KotlinSdk/src/main/kotlin/com/pubnub/apikt/models/consumer/pubsub/PNMessageResult.kt
package com.pubnub.apikt.models.consumer.pubsub

import com.google.gson.JsonElement
import com.pubnub.apikt.callbacks.SubscribeCallback

/**
 * Wrapper around an actual message received in [SubscribeCallback.message].
 */
data class PNMessageResult internal constructor(
    private val basePubSubResult: PubSubResult,
    override val message: JsonElement
) : MessageResult, PubSubResult by basePubSubResult
